### PR TITLE
refactor(compiler): refactor and export Bazel host creation.

### PR DESCRIPTION
**Pull request by dgp1130**
## PR Type
What kind of change does this PR introduce?

- [X] Refactoring (no functional changes, no api changes)

Issue Number: N/A

## What is the new behavior?

Refactored Bazel host creation to export some of the relevant utility functions. google3 needs to make some modifications to the Bazel host to enable the Ivy migration. The functionality exported here is still needed in g3, but it allows g3 code to create the Bazel host, make necessary modifications, and then provide it to the external codebase here while everything is still compatible. For external users, this is a no-op.

## Does this PR introduce a breaking change?

- [X] No

### CODE REVIEW

File: packages/bazel/src/ngc-wrapped/index.ts [hunk 1]
```diff
@@ -433,6 +379,68 @@ export function compile({allDepsCompiledWithBazel = true, compilerOpts, tsHost,
   return {program, diagnostics};
 }
 
+/** A module resolver for handling generated files in Bazel. */
+export function generatedFileModuleResolver(
```
Comment (kyliau) [inline packages/bazel/src/ngc-wrapped/index.ts]: Is this purely moving code around, or are there changes to the logic as well?

Comment (dgp1130) [reply packages/bazel/src/ngc-wrapped/index.ts]: This *should* be a pure refactor. I did change the syntax slightly to be consistent will how I'll be doing most of the host wrapping, but I believe it should not affect functionality.

**Comment by dgp1130** [comment]
I also ran a global presubmit of this commit within google3 and only encountered 4 failures, which appear to be flakes.

**Review by kyliau — APPROVED** [review]

**Comment by kyliau** [comment]
Changing the target from patch-only to master & patch. It'll make it easier to cherry pick changes to patch branch in case subsequent PRs need to target patch.

**Comment by dgp1130** [comment]
Sounds good @kyliau. I also forced the google3 tests to pass as the only failures were a couple flakes. Should be ready to merge.

**Review by IgorMinar — APPROVED** [review]
the commit message explaining the reasoning for this change is quite sparse, but the PR has a great description. @dgp1130 can you please update the commit message to contain the explanation from the PR description?

I'm fine with this going to patch branch since it affects only bazel integrations for which we try to keep the diff between master and patch minimal

thanks!

**Comment by dgp1130** [comment]
Updated commit message to include the description from this PR.

**Comment by atscott** [comment]
[presubmit](https://test.corp.google.com/ui#id=OCL:289127981:BASE:289699733:1579031725440:fefc88a9)

**Pull request closed without merging**